mlflow/sklearn/__init__.py

```diff
@@ -455,6 +460,323 @@ def load_model(model_uri):
+     avoid reference to user dataset variables or model variables.
+    """
+    def __init__(self):
+        self._pred_result_id_to_dataset_name_and_run_id = {}
+        self._eval_dataset_info_map = defaultdict(lambda: defaultdict(list))
+        self._metric_api_call_info = defaultdict(lambda: defaultdict(list))
+        self._log_post_training_metrics_enabled = True
+        self._metric_info_artifact_need_update = defaultdict(lambda: False)
+
+    def should_log_post_training_metrics(self):
+        """
+        Check whether we should run patching code for autologging post training metrics.
+        This checking should surround the whole patched code due to the safe guard checking,
+        See following note.
+
+        Note: It includes checking `_SklearnTrainingSession.is_active()`, This is a safe guarding
+        for meta-estimator (e.g. GridSearchCV) case:
+          running GridSearchCV.fit, the nested `estimator.fit` will be called in parallel,
+          but, the _autolog_training_status is a global status without thread-safe lock protecting.
+          This safe guarding will prevent code run into this case.
+        """
+        return _SklearnTrainingSession.is_active() and self._log_post_training_metrics_enabled
+
+    def disable_log_post_training_metrics(self):
+
+        class LogPostTrainingMetricsDisabledScope:
+            def __enter__(inner_self):
+                inner_self.old_status = self._log_post_training_metrics_enabled
+                self._log_post_training_metrics_enabled = False
+
+            def __exit__(inner_self, exc_type, exc_val, exc_tb):
+                self._log_post_training_metrics_enabled = inner_self.old_status
+
+        return LogPostTrainingMetricsDisabledScope()
+
+    @staticmethod
+    def get_run_id_for_model(model):
+        return getattr(model, '_mlflow_run_id', None)
+
+    @staticmethod
+    def is_metrics_value_loggable(metric_value):
+        """
+        check whether the specified `metric_value` is a numeric value which can be logged
+        as an MLflow metric.
+        """
+        return isinstance(metric_value, (int, float, np.number)) and \
+            not isinstance(metric_value, (bool, np.bool))
+
+    def register_model(self, model, run_id):
+        """
+        In `patched_fit`, we need register the model with the run_id used in `patched_fit`
+        So that in following metric autologging, the metric will be logged into the registered
+        run_id
+        """
+        model._mlflow_run_id = run_id
+
+    @staticmethod
+    def gen_name_with_index(name, index):
+        assert index >= 0
+        if index == 0:
+            return name
+        else:
+            # Use '-' as the separator between name and index,
+            # The '-' is not valid character in python var name
+            # so it can prevent name conflicts after appending index.
+            return f'{name}-{index + 1}'
+
+    def register_eval_dataset(self, model, eval_dataset):
```

Comment:
I think this naming is clearer:

```suggestion
    def register_prediction_input_dataset(self, model, eval_dataset):
```